### ISSUE TITLE
关于Flops

### ISSUE THREAD

**Issue by WZX2022** [ming053l/DRCT#36]
作者您好，我想请问一下论文中DRCT和HAT模型的Flops计算是基于怎样的输入尺寸呢

**Comment by xduzhangjiayu** [comment]
> 作者您好，我想请问一下论文中DRCT和HAT模型的Flops计算是基于怎样的输入尺寸呢

我发现同样的输入HAT要比DRCT快很多，不知道是不是我配置的问题？而且我发现DRCT保存下来的checkpoint特别大，很奇怪

**Comment by ming053l** [comment]
https://drive.google.com/drive/folders/1emyaw6aQvhFgFC_RjK1Qo9c1sTRr-avk

https://drive.google.com/drive/folders/1W-2EEC5mclFzzWrp65u7JDuLDiaA_vPX

你好，我們在論文中提到的DRCT/HAT/DRCT-L/HAT-L的參數量是參考basicsr的log

我們當初CVPRW的版本中的確實存在錯誤，這是我個人的不慎。

後續我們已更新正確的數值到README以及重新上傳正確的數值到arxiv上，請再參考 https://github.com/ming053l/DRCT
以及正確的arxiv版本
--
當初FLOPs我們都是用3x64x64的影像當作輸入去計算，(我們當初使用thop)

會造成具體差別的原因我並不清楚是哪個層面影響的，很抱歉我不能回答你

之前我有試著去對DRCT的STL進行改進(例如使用相對複雜的window-attention/加入CAB/採用mamba的Block)，但不同設備訓練起來速度差非常多，不曉得是不是設備的影響

Hello,

The parameter counts for DRCT/HAT/DRCT-L/HAT-L mentioned in our paper are
based on the logs from BasicSR.

There was indeed an error in the CVPRW version of our work, which was due
to my personal oversight.

We have since corrected the figures and updated them in the README, as well
as uploaded the correct values to the arXiv version. Please refer to
https://github.com/ming053l/DRCT and the corrected arXiv version for
accurate information.

**Comment by xduzhangjiayu** [comment]
@ming053l 作者您好,
我使用这个工程去训练Real-DRCT-L-GANx4模型，发现保存下来的checkpoints有463MB大小，远远大于模型27M的参数，请问这是为什么呢？
另外我从这个项目的主页的表格来看，DRCT是要比HAT快的，但是从我实测来看HAT是要比DRCT快很多的。

期待您可以解答我的疑惑，谢谢~

**Comment by ming053l** [comment]
Regarding the complexity data in DRCT's official repo, it's based on the
log generated by BasicSR package after training, which might be the main
cause of your concern about this.

Here's the data comparison for the DRCT-L model:

   1. Parameters reported by BasicSR (27,285,295):


   - This number is calculated through model.parameters()
   - Only counts parameters that need training (requires_grad=True)
   - Doesn't include parameters that don't need training (like running
   mean/variance in batch normalization)
   - This number represents the actual parameters involved in training


   2. params_ema parameters (37,508,911):


   - This includes all parameters in the model
   - Includes parameters that don't need training (like batch normalization
   statistics)
   - May include some additional buffers
   - EMA version saves moving averages of all parameters

You can refer to the above regarding concerns about file size and model
size.

As for reading speed, in our experiments, DRCT is faster than HAT. However,
due to dense connection requiring storage of all feature map calculation
results within RDG, this significantly tests memory capacity and cache
access speed. This might be one reason for the difference, but I'm not
entirely certain due to equipment inconsistency (which is hard to
standardize) - these are more from my own observations.

In our paper, we used the following equipment specifications for
experiments: CPU: Intel(R) Xeon(R) Gold 6248R CPU @ 3.00GHz GPU: NVIDIA RTX
3090x4 RAM: 442GB

When replacing DRCT components, we tried introducing Mamba's VSSM
architecture, but training and inference speeds were far below
expectations. Later, we saw other researchers integrating Mamba into the
DRCT architecture, and it seemed quite fast, which is one basis for my
speculation that speed inconsistencies come from equipment differences.
This is for your reference.

captainzz ***@***.***> 於 2025年1月16日 週四 下午6:18寫道：

**Comment by WZX2022** [comment]
@ming053l 非常感谢您的回复！
